Homeworks/Homework_5/part_4.py

```diff
@@ -0,0 +1,173 @@
+"""
+The module contains functions for querying and analyzing banking-related data from the database.
+It focuses on generating insights such as assigning discounts, retrieving user information, analyzing bank statistics,
+and filtering transaction data.
+
+Functions:
+- randomly_assign_discounts(cursor): Randomly assigns discounts to a specified number of users.
+- get_user_name_and_surname(cursor, debtor_id): Retrieve the name and surname of a user (debtor) based on their user ID.
+- get_debtors_full_name(cursor): Retrieves the full names of users with negative account balances.
+- get_bank_with_the_biggest_capital(cursor): Retrieves the bank with the highest capital in converted currency.
+- get_bank_with_the_oldest_client(cursor): Retrieves the name of the bank with the oldest client.
+- get_bank_with_highest_unique_outbound_users(cursor): Retrieves the bank with the highest number of unique users
+  with outbound transactions.
+- filter_by_datetime_transactions(transaction): Filters transactions based on their datetime.
+- get_user_last_three_months_transactions(cursor, user_id): Retrieves a user's transactions from the last three months.
+
+Global Constants:
+- logger: The logger instance configured for the module.
+
+Note: The module uses functions from the "utils" module for database connection management, as well as
+global constants and SQL commands from the "constants" module. It also utilizes the "groupby" function from the
+"itertools" module for grouping transactions by bank name.
+"""
```

Comment:
great! it makes the module much more understandable, but you need to know -- it's not necessary. 
but doc strings for classes and functions are